src/config-import.js

```diff
@@ -0,0 +1,246 @@
+     */
+    async *icons() {
+      const iconEntries = entries
+        .filter(
+          (e) =>
+            // omit the icons directory itself
+            e.filename.startsWith('icons/') && !e.filename.endsWith('icons/')
+        )
+        .sort()
+      /** @type {IconData | undefined} */
+      let icon
+      for (const entry of iconEntries) {
+        const buf = await buffer(await entry.openReadStream())
+        const iconFilename = entry.filename.replace(/^icons\//, '')
+        try {
+          const { name, variant } = parseIcon(iconFilename, buf)
+          if (!icon) {
+            icon = {
+              name,
+              variants: [variant],
+            }
+          } else if (icon.name === name) {
+            icon.variants.push(variant)
+          } else {
+            yield icon
+            icon = {
+              name,
+              variants: [variant],
+            }
+          }
+        } catch (err) {
+          warnings.push(err)
+        }
+      }
+      if (icon) {
+        yield icon
+      }
+    },
+
+    /**
+     * @returns {Iterable<{ name: string, value: import('@mapeo/schema').FieldValue }>}
+     */
+    *fields() {
+      const { fields } = presetsFile
+      for (const [name, field] of Object.entries(fields)) {
+        if (typeof field !== 'object' || field === null || !('key' in field)) {
+          warnings.push(new Error(`Invalid field ${name}`))
+          continue
+        }
+        /** @type {any} */
+        const fieldValue = {
+          schemaName: 'field',
+          tagKey: field.key,
+        }
+        for (const key of Object.keys(valueSchemas.field.properties)) {
+          if (key in field) {
+            // @ts-ignore - we validate below
+            fieldValue[key] = field[key]
+          }
+        }
+        if (!validate('field', fieldValue)) {
+          warnings.push(new Error(`Invalid field ${name}`))
+          continue
+        }
+        yield {
+          name,
+          value: fieldValue,
+        }
+      }
+    },
+
+    /**
+     * @returns {Iterable<{ fieldNames: string[], iconName: string | undefined, value: import('@mapeo/schema').PresetValue }>}
+     */
+    *presets() {
+      const { presets } = presetsFile
+      // sort presets using the sort field, turn them into an array
+      const sortedPresets = Object.keys(presets)
+        .filter((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          const isInvalidPreset = typeof preset !== 'object' || preset === null
+          if (isInvalidPreset) {
+            warnings.push(new Error(`invalid preset ${presetName}`))
+          }
+          return !isInvalidPreset
+        })
+        .map((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          if (!preset.sort) {
+            // if there's no sort field, put a big value - puts the field at the end -
+            preset.sort = 100
```

Comment:
Nit: could we use `Infinity` instead of `100`?

```suggestion
            preset.sort = Infinity
```

Reply:
Wouldn't this cause an issue with JSON since `Infinity` is not supported? (I don't remember is JSONSchema `JSON.stringify`'s stuff)

Reply:
I actually think we don't want `sort` on the final output at all, so maybe we could onnly define it in `.sort`? See my suggested code below for my take.

(I might be wrong—maybe we _do_ want sort! If so, `Infinity` gets stringified as `null`, which is probably not what we want.)

Reply:
Yeah, you're right, we don't need sort on the output, since we return the presets already sorted. The use for this is when calling `this.$setProjectSettings` and we pass a list of presetIds, the order determines the sorting (see [here](https://github.com/digidem/mapeo-core-next/blob/8e31ab08dccbaeb0a78ff32a0e0867b321fd658c/src/mapeo-project.js#L691))

Reply:
(I don't see any suggested code below :|), but I've committed some changes:
1. basically define sort as 
```js
const sort = preset.sort || Infinity
const nextSort = nextPreset.sort || Infinity
```
on the sort fn
2. `delete preset.sort` on a `.map` afterwards (I don't think this is strictly necessary, just a nice-to-do)